packages/adapter-vercel/files/serverless.js

```diff
@@ -35,12 +35,16 @@ export default async (req, res) => {
 
 	const request = await getRequest({ base: `https://${req.headers.host}`, request: req });
 
-	setResponse(
-		res,
-		await server.respond(request, {
-			getClientAddress() {
-				return /** @type {string} */ (request.headers.get('x-forwarded-for'));
-			}
-		})
-	);
+	// Get the response from the server
+	const response = await server.respond(request, {
+		getClientAddress() {
+			return /** @type {string} */ (request.headers.get('x-forwarded-for'));
+		}
+	});
+
+	if (req.url?.startsWith(`/${manifest.appPath}/immutable/`) && response.status !== 200) {
+		response.headers.set('cache-control', 'no-store');
```

Comment:
Might be more appropriate
```suggestion
		response.headers.set('cache-control', 'public, max-age=0, must-revalidate');
```